teuthology/suite/run.py

```diff
@@ -501,6 +501,14 @@ def schedule_jobs(self, jobs_missing_packages, jobs_to_schedule, name):
                 log.info("pause between jobs : --throttle " + str(throttle))
                 time.sleep(int(throttle))
 
+    def check_priority(self, jobs_to_schedule, name):
+        priority = self.args.priority
+        if priority < 50:
+            util.schedule_fail('This priority is used only in emergency. To force run, pass --skip', name)
+        elif priority < 75 and jobs_to_schedule > 25:
+            util.schedule_fail('To be used for running less than 25 jobs. To force run, pass --skip', name)
+        elif priority < 150 and jobs_to_schedule > 100:
+            util.schedule_fail('To be used for running less than 100 jobs. To force run, pass --skip', name)
```

Comment:
What do you think about adding a reference to the message with detailed description how priorities should be used, like you've added to the commit message:

Or just print whole hint:
```Priority < 10: Use this if the sky is falling and some group of tests must be run ASAP.

10 <= Priority < 50: Use this if your tests are urgent and blocking other important development.

50 <= Priority < 75: Use this if you are testing a particular feature/fix and running fewer than about 25 jobs. This range can also be used for urgent release testing.

75 <= Priority < 100: Tech Leads will regularly schedule integration tests with this priority to verify pull requests against master.

100 <= Priority < 150: This priority is to be used for QE validation of point releases.

150 <= Priority < 200: Use this priority for 100 jobs or fewer of a particular feature/fix that you’d like results on in a day or so.

200 <= Priority < 1000: Use this priority for large test runs that can be done over the course of a week.```

Reply:
I think we should also update https://docs.ceph.com/docs/master/dev/developer_guide/tests-integration-tests/#testing-priority to reflect this change.